pkg/installer/aromanifests.go

```diff
@@ -0,0 +1,114 @@
+package installer
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/ignition"
+	"github.com/openshift/installer/pkg/asset/ignition/bootstrap"
+	"github.com/pkg/errors"
+
+	"github.com/openshift/installer-aro-wrapper/pkg/cluster/graph"
+)
+
+const (
+	aroManifestDir = "manifests"
+	rootPath       = "/opt/openshift"
+)
+
+// Custom ARO asset to add custom manifests to install graph in installer-wrapper similar to installer's manifests.Manifests
+type AROManifests struct {
+	FileList []*asset.File
+}
+
+// ARO File Fetcher to read manifests
+type aroFileFetcher struct {
+	directory string
+}
+
+var (
+	_ asset.WritableAsset = (*AROManifests)(nil)
+	_ asset.FileFetcher   = (*aroFileFetcher)(nil)
+)
+
+func (am *AROManifests) Name() string {
+	return "ARO Manifests"
+}
+
+func (am *AROManifests) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+func (am *AROManifests) Generate(dependencies asset.Parents) error {
+	return nil
+}
+
+func (am *AROManifests) Files() []*asset.File {
+	return am.FileList
+}
+
+func (am *AROManifests) Load(f asset.FileFetcher) (found bool, err error) {
+	yamlFileList, err := f.FetchByPattern(filepath.Join(aroManifestDir, "*.yaml"))
+	if err != nil {
+		return false, errors.Wrap(err, "failed to load *.yaml files")
+	}
+	ymlFileList, err := f.FetchByPattern(filepath.Join(aroManifestDir, "*.yml"))
+	if err != nil {
+		return false, errors.Wrap(err, "failed to load *.yml files")
+	}
+
+	am.FileList = append(am.FileList, yamlFileList...)
+	am.FileList = append(am.FileList, ymlFileList...)
+	asset.SortFiles(am.FileList)
+
+	return len(am.FileList) > 0, nil
+}
+
+// Append ARO manifest files to the generated graph's bootstrap asset
+func (am *AROManifests) AppendFilesToBootstrap(g graph.Graph) error {
+	bootstrap := g.Get(&bootstrap.Bootstrap{}).(*bootstrap.Bootstrap)
+	for _, file := range am.Files() {
+		manifest := ignition.FileFromBytes(filepath.Join(rootPath, file.Filename), "root", 0644, file.Data)
```

Comment:
By doing this, do we even _need_ this struct to be an Asset? It looks like the actual process of loading the manifests would then be done by the bootstrap asset, not this one?

Reply:
We are not using aroManifests exactly similar as manifests.Manifests here, because aroManifests has to be under dependencies of some other asset's dependency such that the files go into bootstrap.Bootstrap{} when actual graph creation happens.
I think we should still replicate and keep using aroManifests as an asset because it is keeping it very similar to manifests.Manifests{}, only differing by where it is getting added to the graph.